src/doc_table.c

```diff
@@ -317,6 +317,14 @@ RSDocumentMetadata *DocTable_Pop(DocTable *t, const char *s, size_t n) {
 
     md->flags |= Document_Deleted;
 
+    t->memsize -= sizeof(RSDocumentMetadata) + sdsAllocSize(md->keyPtr);
```

Comment:
@ashtul are you sure we are not updating those stats somewhere else?

Reply:
Yes